lib/helpers.js

```diff
@@ -695,6 +695,39 @@ async function formatConfigMarker (configsGetter, desiredMarker, defaultMarker)
   return configMarker;
 }
 
+/**
+ * Parses the output in JSON format retrieved from
+ * the corresponding Appium Settings broadcast calls
+ *
+ * @param {string} output The actual command output
+ * @param {string} entityName The name of the entity which is
+ * going to be parsed
+ * @returns {Object} The parsed JSON object
+ * @throws {Error} If the output cannot be parsed
+ * as a valid JSON
+ */
+function parseJsonData (output, entityName) {
+  if (!/\bresult=-1\b/.test(output) || !/\bdata="/.test(output)) {
+    log.debug(output);
+    throw new Error(`Cannot retrieve ${entityName} from the device. ` +
+      'Check the server log for more details');
+  }
+  const match = /\bdata=(".+)/.exec(output);
+  if (!match) {
```

Comment:
Are there cases where `/\bdata="/.test(output)` will be `true` but `/\bdata=(".+)/.exec(output)` will not match anything?

Reply:
very unlikely